config/prometheus/monitor.yaml

```diff
@@ -17,4 +17,4 @@ spec:
         insecureSkipVerify: true
   selector:
     matchLabels:
-      control-plane: glance-controller-manager
+      control-plane: controller-manager
```

Comment:
AFAIK you need to have unique newly added label here too for the endpoint..

Reply:
I haven't seen any salient problem in my testing, but what you're saying makes sense.

Reply:
I think for now I would opt not to change this here or with `config/rbac/auth_proxy_service.yaml`, since they pre-existed webhooks (see Nova operator [1][2] as an example, which does not yet have webhooks) and seem to have been working fine so far.  Perhaps, however, they just haven't been properly tested.  Anyhow, I'd just like to deal with the webhooks in this patch and we can follow-up with other PRs for these other label selectors.

[1] https://github.com/openstack-k8s-operators/nova-operator/blob/master/config/prometheus/monitor.yaml#L18-L20
[2] https://github.com/openstack-k8s-operators/nova-operator/blob/aa6184d9b90492a975f56c1791994072c48ad824/config/rbac/auth_proxy_service.yaml#L14-L15

Reply:
Not sure what "working fine means here" :smile:  The issue is not webhook specific as I mentioned in https://issues.redhat.com/browse/OSP-23156. AFAICT we don't test them and don't see the issue. Also, this change is reverting something which fixed it earlier.

Reply:
> Also, this change is reverting something which fixed it earlier.

In the context of webhooks, it is indeed reverting the `<operator name>-controller-manager` labeling, but it is not "unfixing" the functionality in that regard.  The webhook traffic will still be routed properly with the new `app.kubernetes.io/name: glance-operator` label and label selector.  I have tested both `Glance` and `Keystone` operator deployed at the same time with these new labels, and their respective webhooks properly received traffic (I repeated the test over and over so as to rule-out that I was "just getting lucky").

As I mentioned above regarding the Prometheus and RBAC concerns, "`Perhaps, however, they just haven't been properly tested`".  We should figure out what that testing would look like.

Reply:
I wonder if we could just use the new `app.kubernetes.io/name: <operator name>-operator` label here [1] and here [2].

[1] https://github.com/abays/glance-operator/blob/30ad3b2abf249cba2ec93235fc8323ccdea09d25/config/rbac/auth_proxy_service.yaml#L14-L15
[2] https://github.com/abays/glance-operator/blob/30ad3b2abf249cba2ec93235fc8323ccdea09d25/config/prometheus/monitor.yaml#L18-L20

Reply:
>  but it is not "unfixing" the functionality in that regard

Well, we're assuming the earlier change was `only` in the context of webhooks, which is not correct.  I changed it for all services including webhook service.  Though I did not change in all operators.

Reply:
> I wonder if we could just use the new app.kubernetes.io/name: <operator name>-operator label here [1] and here [2].

We could/should and that's what I was suggesting in my comment I think. They have to be unique labels for the operator deployments.

Reply:
> > but it is not "unfixing" the functionality in that regard
> 
> Well, we're assuming the earlier change was `only` in the context of webhooks, which is not correct. I changed it for all services including webhook service. Though I did not change in all operators.

That's not what I said.  `"In the context of webhooks, it is indeed..."`.  I was speaking of webhooks.  Then later I mentioned `"...regarding the Prometheus and RBAC concerns"`, which, yes, is another concern affected by these labels.  But that is something we need to qualify with a proper test so to understand how to determine what the proper approach is here.

Reply:
> > I wonder if we could just use the new app.kubernetes.io/name: -operator label here [1] and here [2].
> 
> We could/should and that's what I was suggesting in my comment I think. They have to be unique labels for the operator deployments.

Okay, I will change those and see what happens.  And we can follow-up with investigating how to properly test the Prometheus/RBAC stuff.